### PR TITLE
[RPA-1309] Export to HTML action now keeps cell formatting

### PR DESCRIPTION
**Pull request by bhszewczyk**
Previously exporting to HTML caused cell formatting to be converted as value:

Now it takes the final outcome (formatting do not mess with value):

Excel:
image

HTML:
image

### CODE REVIEW

File: office-actions/src/automation/excel/excel.action-handler.ts [hunk 1]
```diff
@@ -343,29 +343,31 @@ export default class ExcelActionHandler extends StatefulActionHandler {
...
-                rangeValues.push(rowValues);
+        const startCell = splitCellRange[0]
+        const endCell = splitCellRange[1]
+        const { column: startColumn, row: startRow } = this.getDividedCellCoordinates(startCell);
+        const { column: endColumn, row: endRow } = this.getDividedCellCoordinates(endCell);
+
+        for (let rowIdx = startRow; rowIdx <= endRow; rowIdx++) {
+            const rowValues: ExcelCellValue[] = [];
+            for (let columnIdx = startColumn; columnIdx <= endColumn; columnIdx++) {
+                console.log(`${rowIdx} ${columnIdx}:`, targetWorksheet.Cells(rowIdx, columnIdx).Text ?? '');
```
Comment (marcin-depcik) [inline office-actions/src/automation/excel/excel.action-handler.ts]: is this `console.log` was intentional?

File: office-actions/src/automation/excel/excel.action-handler.ts [hunk 2]
```diff
@@ -343,29 +343,31 @@ export default class ExcelActionHandler extends StatefulActionHandler {
...
 
         const targetWorksheet = this.session.Worksheets(input?.worksheet ?? this.session.ActiveSheet.Name);
-
         const rowsRange = targetWorksheet.Range(input.cellRange).Rows;
         const rowsCount = rowsRange.Count;
 
         if (input.rowLevel && Number(input.rowLevel) <= 0) {
             throw new Error(ExcelErrorMessage.createHtmlTableInvalidRowLevel());
         }
-        const rowLevel = input.rowLevel ? Number(input.rowLevel) : 1;
```
Comment (marcin-depcik) [inline office-actions/src/automation/excel/excel.action-handler.ts:353]: why?

**Review by marcin-depcik — CHANGES_REQUESTED** [review]
bump version

File: office-actions/src/automation/excel/excel.action-handler.ts (hunk 2 above)
Comment (bhszewczyk) [reply office-actions/src/automation/excel/excel.action-handler.ts:353]: cause i do not use this variable anymore

File: office-actions/src/automation/excel/excel.action-handler.ts [hunk 3]
```diff
@@ -359,15 +359,25 @@ export default class ExcelActionHandler extends StatefulActionHandler {
...
+        const { column: startColumn, row: startRow } = this.getDividedCellCoordinates(startCell);
+        const { column: endColumn, row: endRow } = this.getDividedCellCoordinates(endCell);
 
-            if (rowOutlineLevel <= rowLevel) {
-                rangeValues.push(rowValues);
+        for (let rowIdx = startRow; rowIdx <= endRow; rowIdx++) {
+            const rowOutlineLevel = rowsRange.Rows(rowIdx).OutlineLevel;
+            if (rowOutlineLevel > rowLevel) {
+                continue;
+            };
```
Comment (marcin-depcik) [inline office-actions/src/automation/excel/excel.action-handler.ts]: what do you think?
`if (rowOutlineLevel > rowLevel) continue;`

**Review by marcin-depcik — APPROVED** [review]

File: office-actions/src/automation/excel/excel.action-handler.ts [hunk 4]
```diff
@@ -359,15 +359,25 @@ export default class ExcelActionHandler extends StatefulActionHandler {
...
+            const rowValues: ExcelCellValue[] = [];
+         
+            for (let columnIdx = startColumn; columnIdx <= endColumn; columnIdx++) {
+                rowValues.push(targetWorksheet.Cells(rowIdx, columnIdx).Text ?? '');
             }
+
+            rangeValues.push(rowValues);
         }
 
+        console.log(rangeValues);
```
Comment (marcin-depcik) [inline office-actions/src/automation/excel/excel.action-handler.ts]: remove `console.log`

**Pull request merged**